pkg/policy/k8s/service.go

```diff
@@ -56,8 +40,7 @@ func (p *policyWatcher) onServiceEvent(event k8s.ServiceNotification) {
 func (p *policyWatcher) updateToServicesPolicies(svcID k8s.ServiceID, newSVC, oldSVC *k8s.Service) error {
 	var errs []error
 
-	// Bail out early if updated service is not selectable
-	if !(p.isSelectableService(newSVC) || p.isSelectableService(oldSVC)) {
+	if newSVC == nil {
```

Comment:
Are we 100% sure this is safe? By reading this, my intuition says that this line "ignores" delete events. eg. - but I don't think that is the case. Looking a bit below, the existing line `newSVC.DeepEqual(oldSVC)` implicitly says that newSVC is always non-nil, because otherwise it will be a null pointer panic. I defer to other cilium maintainers how to deal with this, but it seems strange to have a silent`if newSVC == nil` if we never send in `nil`, and I think it in the future can hide resource leaks.